gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GitFlowGraphMonitor.java

```diff
@@ -0,0 +1,332 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.service.modules.core;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.gobblin.util.reflection.GobblinConstructorUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.eclipse.jgit.diff.DiffEntry;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import com.google.common.io.Files;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import com.typesafe.config.ConfigValueFactory;
+
+import org.apache.gobblin.configuration.ConfigurationKeys;
+import org.apache.gobblin.service.modules.flowgraph.DataNode;
+import org.apache.gobblin.service.modules.flowgraph.FlowEdge;
+import org.apache.gobblin.service.modules.flowgraph.FlowEdgeFactory;
+import org.apache.gobblin.service.modules.flowgraph.FlowGraph;
+import org.apache.gobblin.service.modules.flowgraph.FlowGraphConfigurationKeys;
+import org.apache.gobblin.service.modules.template_catalog.FSFlowCatalog;
+import org.apache.gobblin.util.ConfigUtils;
+import org.apache.gobblin.util.PullFileLoader;
+
+import joptsimple.internal.Strings;
+import lombok.extern.slf4j.Slf4j;
+
+
+/**
+ * Service that monitors for changes to {@link org.apache.gobblin.service.modules.flowgraph.FlowGraph} from a git repository.
+ * The git repository must have an inital commit that has no files since that is used as a base for getting
+ * the change list.
+ * The {@link DataNode}s and {@link FlowEdge}s in FlowGraph need to be organized with the following directory structure on git:
+ * <root_flowGraph_dir>/<nodeName>/<nodeName>.properties
+ * <root_flowGraph_dir>/<nodeName1>/<nodeName2>/<edgeName>.properties
+ */
+@Slf4j
+public class GitFlowGraphMonitor extends GitMonitoringService {
+  private static final int NODE_FILE_DEPTH=3;
+  private static final int EDGE_FILE_DEPTH=4;
+  private static final String PROPERTIES_EXTENSIONS = "properties,props";
+  private static final String CONF_EXTENSIONS = "configuration,conf";
+  public static final String FLOW_EDGE_LABEL_JOINER_CHAR = ":";
+
+  private FSFlowCatalog flowCatalog;
+  private FlowGraph flowGraph;
+  private String flowGraphDir;
+  private String repositoryDir;
+  private Path flowGraphDirPath;
+  private final PullFileLoader pullFileLoader;
+  private final Config emptyConfig = ConfigFactory.empty();
+  private final Set<String> javaPropsExtensions;
+  private final Set<String> hoconFileExtensions;
```

Comment:
When both hocon and properties exist in gitRepo, which one will take priority for loading inside of flowCompiler?

Reply:
Right now, the one that comes later will take the priority i.e. the one that comes later will result in a new FlowEdge that will overwrite the previous FlowEdge.

Reply:
See comment above. To eliminate non-deterministic behavior introduced by different files with same file names and different file extensions, I have changed the code to support only ".properties" files.